react/features/dial-out/components/DialOutDialog.web.js

```diff
@@ -0,0 +1,226 @@
+     */
+    render() {
+        const { _isDialNumberAllowed } = this.props;
+
+        return (
+            <Dialog
+                okDisabled = { this.state.isDialInputEmpty
+                    || !_isDialNumberAllowed }
+                okTitleKey = 'dialOut.dial'
+                onCancel = { this._onCancel }
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialOut.dialOut'
+                width = 'small'>
+                { this._renderContent() }
+            </Dialog>);
+    }
+
+    /**
+     * Formats the dial number in a way to remove all non digital characters
+     * from it (including spaces, brackets, dash, dot, etc.).
+     *
+     * @param {string} dialNumber - The phone number to format.
+     * @private
+     * @returns {string} - The formatted phone number.
+     */
+    _formatDialNumber(dialNumber) {
+        return dialNumber.replace(/\D/g, '');
+    }
+
+    /**
+     * Renders the dialog content.
+     *
+     * @returns {XML}
+     * @private
+     */
+    _renderContent() {
+        const { _isDialNumberAllowed } = this.props;
+
+        return (
+            <div className = 'dial-out-content'>
+                { _isDialNumberAllowed ? '' : this._renderErrorMessage() }
+                <DialOutNumbersForm
+                    onChange = { this._onDialNumberChange } />
+            </div>);
+    }
+
+    /**
+     * Renders the error message to display if the dial phone number is not
+     * allowed.
+     *
+     * @returns {XML}
+     * @private
+     */
+    _renderErrorMessage() {
+        const { t } = this.props;
+
+        return (
+            <div className = 'dial-out-error'>
+                { t('dialOut.phoneNotAllowed') }
+            </div>);
+    }
+
+    /**
+     * Cancel the dial out.
+     *
+     * @private
+     * @returns {boolean} - Returns true to indicate that the dialog should be
+     * closed.
+     */
+    _onCancel() {
+        this.props.cancel();
+
+        return true;
+    }
+
+    /**
+     * Dials the number.
+     *
+     * @private
+     * @returns {boolean} - Returns true to indicate that the dialog should be
+     * closed.
+     */
+    _onSubmit() {
+        if (this.props._isDialNumberAllowed) {
+            this.props.dial(this.state.dialNumber);
+        }
+
+        return true;
+    }
+
+    /**
+     * Updates the dialNumber and check for validity.
+     *
+     * @param {string} dialCode - The dial code value.
+     * @param {string} dialInput - The dial input value.
+     * @private
+     * @returns {void}
+     */
+    _onDialNumberChange(dialCode, dialInput) {
+        // We remove all starting zeros from the dial input before attaching it
+        // to the country code.
+        const formattedDialInput = dialInput.replace(/^(0+)/, '');
+
+        const dialNumber = `${dialCode}${formattedDialInput}`;
+
+        const formattedNumber = this._formatDialNumber(dialNumber);
+
+        this.props.checkDialNumber(formattedNumber);
```

Comment:
Ah, another issue I saw is that the keypresses will get propagated causing other keyboard shortcuts to fire. That's not really an problem for the alphabet but numbers do lead to focusing on filmstrip thumbnails.